### PR TITLE
Add hamming/instructions.append with exception raising details

### PR DESCRIPTION
**Pull request by IsaacG**

### CODE REVIEW

File: exercises/practice/hamming/.docs/instructions.append.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Implementation notes: exceptions
+
+This exercise expects the code to raise exceptions in certain circumstances.
+Exceptions are usually raised when an operation cannot be completed for some
+reason and there is no value that can be returned which would be correct.
+Instead, an exception is raised. When raising an exception, the tests expect
```
Comment (SleeplessByte) [inline exercises/practice/hamming/.docs/instructions.append.md:6]: Doesn't seem to be an "instead" clause as the previous sentence already says there is no clause. `Therefore` might be more applicable.

File: exercises/practice/hamming/.docs/instructions.append.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Implementation notes: exceptions
+
+This exercise expects the code to raise exceptions in certain circumstances.
+Exceptions are usually raised when an operation cannot be completed for some
```
Comment (SleeplessByte) [inline exercises/practice/hamming/.docs/instructions.append.md:4]: FWIW, general markdown style on Exercism is once sentence per line, not limited to a width. This makes reviews, edits, and diffs easier!

File: exercises/practice/hamming/.docs/instructions.append.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+reason and there is no value that can be returned which would be correct.
```
Comment (SleeplessByte) [inline exercises/practice/hamming/.docs/instructions.append.md:5]: I think you meant: and there is no applicable value to return?

"which would be correct" is amtrigous. Either the act of no value returning is correct, or returning a value would be correct, or a correct value cannot be returned.

File: exercises/practice/hamming/.docs/instructions.append.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+Instead, an exception is raised. When raising an exception, the tests expect
+that the exception class in instantiated and contains a message.
+
+```python
+# Avoid this. There is no message and the tests will fail:
+raise ValueError
+# Do this. Provide a clear message with as much detail as possible:
```
Comment (SleeplessByte) [inline exercises/practice/hamming/.docs/instructions.append.md:12]:
```suggestion
# Do this. Provide a clear and actionable message:
```

I don't think we always want to go into detail, but whatever we do, there must be "just" enough information to take an action. A large assssss error message which you can't solve isn't as good as a short, solvable, error. Right?

**Review by SleeplessByte — COMMENTED** [review]
My 2 cents!

File: exercises/practice/hamming/.docs/instructions.append.md (hunk 3 above)
Comment (IsaacG) [reply exercises/practice/hamming/.docs/instructions.append.md:5]: Operation cannot be performed. Function usually returns a value. Any value returned would be incorrect. There exists no value where returning that value would be correct.

Maybe,
> Exceptions are usually raised when an operation cannot be completed for some reason and therefore there is no value to return. Any value returned would be an incorrect value.

File: exercises/practice/hamming/.docs/instructions.append.md (hunk 1 above)
Comment (IsaacG) [reply exercises/practice/hamming/.docs/instructions.append.md]: Changed.

File: exercises/practice/hamming/.docs/instructions.append.md (hunk 4 above)
Comment (IsaacG) [reply exercises/practice/hamming/.docs/instructions.append.md]: Changed.

File: exercises/practice/hamming/.docs/instructions.append.md (hunk 3 above)
Comment (SleeplessByte) [reply exercises/practice/hamming/.docs/instructions.append.md]: Yeah much better!

**Comment by IsaacG** [comment]
Related to https://github.com/exercism/python/issues/2563 (claimed)

**Comment by BethanyG** [comment]
@IsaacG  - Thank you for submitting this, but I am going to close this at this time.  Given our (lengthy) slack discussions, the number of exercises we may now have to change, and the wording in #2563 (which needs to be discussed), I think we're not read to go ahead with changes.

**Pull request closed without merging**